aptos-move/framework/aptos-stdlib/sources/experimental/optional_aggregator.move

```diff
@@ -0,0 +1,190 @@
+/// This module provides an interface to aggregate integers either via
+/// aggregator (parallelizable) or via normal integers.
+module aptos_std::optional_aggregator {
+    use std::error;
+    use std::option::{Self, Option};
+
+    use aptos_std::aggregator_factory;
+    use aptos_std::aggregator::{Self, Aggregator};
+
+    // These error codes are produced by `Aggregator` and used by `Integer` for
+    // consistency.
+    const EAGGREGATOR_OVERFLOW: u64 = 1;
+    const EAGGREGATOR_UNDERFLOW: u64 = 2;
+
+    /// Wrapper around integer to have a custom overflow limit. Note that
+    /// Move has no traits (and trait bounds), so integer value must be u128.
+    /// `Integer` provides API to add/subtract and read, just like `Aggregator`.
+    struct Integer has store {
+        value: u128,
+        limit: u128,
+    }
+
+    fun new_integer(limit: u128): Integer {
+        Integer {
+            value: 0,
+            limit,
+        }
+    }
+
+    fun add_integer(base: &mut Integer, value: u128) {
+        assert!(
+            base.limit >= base.value && value <= (base.limit - base.value),
```

Comment:
It seems that `base.limit >= base.value` is always true, so unnecessary to check. No function including `new_integer` and `add_integer` can break this property (or invariant).